packages/netsuite-adapter/src/data_elements/multi_fields_identifiers.ts

```diff
@@ -13,38 +13,80 @@
-import { BuiltinTypes, Field, ObjectType } from '@salto-io/adapter-api'
+import { BuiltinTypes, Field, ObjectType, Values } from '@salto-io/adapter-api'
+import { logger } from '@salto-io/logging'
 import { values } from '@salto-io/lowerdash'
 import _ from 'lodash'
+import { getDataInstanceId } from '../elements_source_index/elements_source_index'
+import { IDENTIFIER_FIELD, TYPE_TO_IDENTIFIER, TYPE_TO_ID_FIELD_PATHS } from './types'
 
-// This is used for constructing a unique identifier for data types
-// field using multiple other fields
-export const TYPE_TO_ID_FIELD_PATHS: Record<string, string[][]> = {
-  AccountingPeriod: [['periodName'], ['fiscalCalendar', 'name']],
-  Nexus: [['country'], ['state', 'name']],
-  Account: [['acctName'], ['acctNumber']],
-}
+const log = logger(module)
 
-export const IDENTIFIER_FIELD = 'identifier'
 
 export const addIdentifierToType = (type: ObjectType): void => {
-  if (!(type.elemID.name in TYPE_TO_ID_FIELD_PATHS)) {
+  if (!(type.elemID.name in TYPE_TO_ID_FIELD_PATHS) && type.fields.parent === undefined) {
     return
   }
 
   type.fields[IDENTIFIER_FIELD] = new Field(type, IDENTIFIER_FIELD, BuiltinTypes.SERVICE_ID)
 }
 
-export const addIdentifierToValues = (
-  record: Record<string, unknown>,
+const getIdentifierWithoutParent = (
+  record: Values,
   type: ObjectType
-): void => {
+): string => {
   if (!(type.elemID.name in TYPE_TO_ID_FIELD_PATHS)) {
-    return
+    return record[TYPE_TO_IDENTIFIER[type.elemID.name]]
   }
 
-  record.identifier = TYPE_TO_ID_FIELD_PATHS[type.elemID.name]
+  return TYPE_TO_ID_FIELD_PATHS[type.elemID.name]
     .map(fieldPath => _.get(record, fieldPath))
     .filter(values.isDefined)
     .join('_')
 }
+
+const getFullIdentifier = (
+  record: Values,
+  type: ObjectType,
+  internalIdToExternalId: Record<string, Values>
+): string => {
+  const currentInstanceId = getIdentifierWithoutParent(record, type)
+  if (record.parent === undefined) {
+    return currentInstanceId
+  }
+
+  const parent = internalIdToExternalId[
+    getDataInstanceId(record.parent.attributes.internalId, type)
+  ]
+  if (parent === undefined) {
+    log.warn(`Could not find parent with id ${record.parent.attributes.internalId} of instance with id ${record.attributes.internalId} of type ${type.elemID.getFullName()}`)
+    return `${record.parent.attributes.internalId}_${currentInstanceId}`
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  return `${getFullIdentifier(parent, type, internalIdToExternalId)}_${currentInstanceId}`
+}
+
+export const addIdentifierToValues = (
+  valuesList: { type: ObjectType; record: Values }[],
+): void => {
+  const valuesListWithIdentifier = valuesList.filter(
+    ({ type }) => type.fields[IDENTIFIER_FIELD] !== undefined
+  )
+
+  const internalIdToExternalId = _(valuesListWithIdentifier)
+    .keyBy(({ type, record }) => getDataInstanceId(record.attributes.internalId, type))
+    .mapValues(({ record }) => record)
+    .value()
```

Comment:
Q: Does this code really creates `internalIdToExternalId`? seems like the value is not only the externalId.
nit: IMO this would be more readable.
```suggestion
  const internalIdToExternalId = Object.fromEntries(
    valuesListWithIdentifier
      .map(({ type, record }) => [getDataInstanceId(record.attributes.internalId, type), record])
  )
```